models/sdn-fabric-0.1.x/openapi.yaml

```diff
@@ -1280,6 +1329,98 @@ paths:
         "201":
           description: created
       summary: POST /switch/{switch-id}/port
+  /sdn-fabric/v0.1.x/{fabric-id}/switch/{switch-id}/port/{cage-number}/{channel-number}/cage-number/values:
+    description: reference to the cage-number of the port in the switch model
+    get:
+      operationId: getSwitch_Port_Cage-number_Values_Leafref
+      responses:
+        "200":
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/LeafRefOptions'
+          description: GET OK 200
+      summary: GET /switch/{switch-id}/port/{cage-number}/{channel-number}/cage-number/values
+        Leafref
+    parameters:
+    - $ref: '#/components/parameters/fabric-id'
+    - content:
+        text/plain; charset=utf-8:
+          schema:
+            type: string
+      description: key {switch-id}
+      in: path
+      name: switch-id
+      required: true
+    - content:
+        text/plain; charset=utf-8:
+          schema:
+            type: string
+      description: key {cage-number}
+      in: path
+      name: cage-number
+      required: true
+    - content:
+        text/plain; charset=utf-8:
+          schema:
+            type: string
+      description: key {channel-number}
+      in: path
+      name: channel-number
+      required: true
+    x-leafref: /switch-model/port/cage-number
+    x-leafref-target-path: port/cage-number
+    x-leafref-top-model: /sdn-fabric/v0.1.x/{fabric-id}/switch-model
```

Comment:
@SeanCondon I have added these information to the leafref-resolver path, to make it easier when generating api.  I could extract them from the openapi spec when generating the api (rocaas tool), but I think it would be easier here. Please let me know if you think otherwise.

Reply:
x-leafref-top-model is the top common model, and the x-leafref-target-path is to iterate over the model and find the leaf values.